code/game/objects/structures/floodlight.dm

```diff
@@ -0,0 +1,322 @@
+/obj/machinery/power/floodlight
+	name = "floodlight"
+	icon = 'icons/obj/floodlight.dmi'
+	icon_state = "floodlight0_hatchopen0"
+	desc = "An industrial floodlight. It has units of power remaining."
+	density = 1
+	anchored = 0
+	var/status = 0
+	var/obj/item/weapon/stock_parts/cell/powerpack
+	var/datum/effect_system/spark_spread/sparks = new
+	var/health = 60
+	var/maxhealth = 60
+	var/broken = 0
```

Comment:
actually go check here and see which vars to remove https://github.com/HippieStationCode/HippieStation13/blob/5462010e5bcc8a61ad5618ffd0f04f07eeb570b0/code/game/machinery/machinery.dm#L96